modules/snooker/src/main/ts/ephox/snooker/api/ResizeBehaviour.ts

```diff
@@ -0,0 +1,149 @@
+  readonly calcMiddleDeltas: (input: number[], previousIndex: number, index: number, nextIndex: number, delta: number, minCellSize: number, relativeSizing: boolean) => number[];
+  readonly calcRightEdgeDeltas: (input: number[], previousIndex: number, index: number, delta: number, minCellSize: number, relativeSizing: boolean) => number[];
+}
+
+const zero = (array: number[]) => Arr.map(array, Fun.constant(0));
+
+const surround = (input: number[], startIndex: number, endIndex: number, results: number[], f: (array: number[]) => number[]) =>
+  f(input.slice(0, startIndex)).concat(results).concat(f(input.slice(endIndex)));
+
+// Clamps negative (rtl) deltas that cause a column/row to be reduced past its min size
+const clampDelta = (sizes: number[], index: number, delta: number, minCellSize: number) => {
+  const newSize = Math.max(minCellSize, sizes[index] + delta);
+  return newSize - sizes[index];
+};
+
+// Preserve the size of the columns/rows and adjust the table size
+const resizeTable = (): ResizeBehaviour => {
+  const calcFixedDeltas = (input: number[], index: number, next: number, delta: number, minCellSize: number) => {
+    if (delta >= 0) {
+      return surround(input, index, next + 1, [ delta, 0 ], zero);
+    } else {
+      const newThis = Math.max(minCellSize, input[index] + delta);
+      const diff = input[index] - newThis;
+      return surround(input, index, next + 1, [ -diff, 0 ], zero);
+    }
+  };
+
+  const calcRelativeDeltas = (input: number[], index: number, delta: number, minCellSize: number) => {
+    // ASSUMPTION: The delta will be a percentage. This may not be correct if other relative sizing is added, so we probably
+    // need a better way to calc the ratio.
+    const ratio = (100 + delta) / 100;
+    const newThis = Math.max(minCellSize, (input[index] + delta) / ratio);
+    return Arr.map(input, (size, idx) => {
+      const newSize = idx === index ? newThis : size / ratio;
+      return newSize - size;
+    });
+  };
+
+  // Calculations for the inner columns/rows
+  const calcLeftEdgeDeltas = (input: number[], index: number, next: number, delta: number, minCellSize: number, relativeSizing: boolean) => {
+    if (relativeSizing) {
+      return calcRelativeDeltas(input, index, delta, minCellSize);
+    } else {
+      return calcFixedDeltas(input, index, next, delta, minCellSize);
+    }
+  };
+
+  const calcMiddleDeltas = (input: number[], _prev: number, index: number, next: number, delta: number, minCellSize: number, relativeSizing: boolean) =>
+    calcLeftEdgeDeltas(input, index, next, delta, minCellSize, relativeSizing);
+
+  const resizeTable = (resizer: TableResizer, delta: number) =>
+    resizer(delta);
+
+  // Calculations for the last column/row resizer
+  // Resizing the last column/row only requires the delta of the last column/row to be calcuated
```

Comment:
This comment is misleading, as that's true for "fixed" size tables, but for relative tables all rows/cols needs to be adjusted.

Reply:
Have adjusted the comments so hopefully they are better now.